tests/unit/processor/test_processor_strategy.py

```diff
@@ -89,3 +93,40 @@ def test_apply_processor_multiple_times_until_no_new_rule_matches(self):
             processor_metrics=mock_metrics,
         )
         assert expected_event == event
+
+    @pytest.mark.parametrize("execution_number", range(5))  # repeat test to ensure determinism
+    def test_strategy_applies_rules_in_deterministic_order(self, execution_number):
+        generic_tree = RuleTree()
+        specific_tree = RuleTree()
+        rule_one_dict = {"filter": "field", "normalize": {"field.one": "copy.one"}}
+        rule_two_dict = {"filter": "NOT field.two", "normalize": {"field.one": "field.two"}}
+        rule_one = NormalizerRule._create_from_dict(rule_one_dict)
```

Comment:
please do not use the Normalizer rule here. It is replaced in near future

Reply:
I changed the test to make use of the generic_adder this led to some problems with the comparison of the rule due to the `base/rule.py` hash and repr functions. I had to add the objects `id` to the hash function to make a unique hash per object. Without that either all test repetitions are always green or red (in the negative case when the bugfix is not present in the processor_strategy).